cli/src/main.rs

```diff
@@ -52,6 +54,225 @@ pub fn main() {
+            OsString::from("bin"),
+        ]
+        .iter()
+        .collect();
+
+        let zcash_confs: PathBuf = [
+            worktree_home.clone(),
+            OsString::from("regtest"),
+            OsString::from("conf"),
+            OsString::from(""),
+        ]
+        .iter()
+        .collect();
+
+        let mut flagged_zcashd_conf: String = "--conf=".to_string();
+        flagged_zcashd_conf.push_str(zcash_confs.to_str().expect("error making zcash_datadir"));
+        flagged_zcashd_conf.push_str("zcash.conf");
+
+        let zcash_datadir: PathBuf = [
+            worktree_home.clone(),
+            OsString::from("regtest"),
+            OsString::from("datadir"),
+            OsString::from("zcash"),
+            OsString::from(""),
+        ]
+        .iter()
+        .collect();
+
+        let mut flagged_datadir: String = "--datadir=".to_string();
+        flagged_datadir.push_str(zcash_datadir.to_str().expect("error making zcash_datadir"));
+
+        let mut zcash_logs: PathBuf = [
+            worktree_home.clone(),
+            OsString::from("regtest"),
+            OsString::from("logs"),
+            OsString::from(""),
+        ]
+        .iter()
+        .collect();
+
+        let mut zcashd_bin = bin_location.to_owned();
+        zcashd_bin.push("zcashd");
+
+        // TODO reorg code, look for all needed bins ASAP
+        // check for file. This might be superfluous considering
+        // .expect() attached to the call, below?
+        if !std::path::Path::is_file(zcashd_bin.as_path()) {
+            panic!("can't find zcashd bin! exiting.");
+        }
+
+        println!("zcashd datadir: {}", &flagged_datadir);
+        println!("zcashd conf file: {}", &flagged_zcashd_conf);
+        let mut zcashd_command = Command::new(zcashd_bin)
+            .args([
+                "--printtoconsole",
+                &flagged_zcashd_conf,
+                &flagged_datadir,
+                // Right now I can't get zcashd to write to debug.log with this flag
+                //"-debuglogfile=.../zingolib/regtest/logs/debug.log",
+                //debug=1 will at least print to stdout
+                "-debug=1",
+            ])
+            // piping stdout off...
+            .stdout(Stdio::piped())
+            .stderr(Stdio::piped())
+            .spawn()
+            .expect("failed to start zcashd");
+
+        // ... to ... here
+        if let Some(mut zcashd_log) = zcashd_command.stdout.take() {
+            std::thread::spawn(move || {
+                let mut zcashd_stdout_log: String = String::new();
+                zcashd_stdout_log.push_str(zcash_logs.to_str().expect("error converting to str"));
+                zcashd_stdout_log.push_str("zcashd_stdout.log");
+
+                let mut zcashd_logfile =
+                    File::create(zcashd_stdout_log).expect("file::create Result error");
+                std::io::copy(&mut zcashd_log, &mut zcashd_logfile)
```

Comment:
Wow...   this...  works?   A mutable reference to a value is passed on the right hand side of the `let` statement that creates the variable?!